### ISSUE TITLE
The "this" value of decorated methods gets changed

### ISSUE THREAD

**Issue by gremo** [nestjs-community/nestjs-config#24]
I think maybe there is a logic error in the way `@Configurable()` works [here](https://github.com/nestjs-community/nestjs-config/blob/master/src/decorators/Configurable.ts#L20): `apply `changes the value of `this`, so your are not able to call any method or property inside a decorated `method1`:

 ```typescript
@Injectable()
class TestService {
  @Configurable()
  method1(@ConfigParam('foo') foo: string) {
    this.method2();
  }

  method2() {
    console.log('method2');
  }
}
```

> TypeError: this.method2 is not a function

I may be wrong; I'm not a guru in JavaScript but I fear we can't solve this. At least, this should be clearly explained in the documentation.

**Comment by gremo** [comment]
Also see this: https://stackoverflow.com/questions/35073205/problems-decorating-an-instance-method-in-javascript

**Comment by shekohex** [comment]
thank you @gremo

**Comment by gremo** [comment]
This will not work with instance variables (and I'm afraid it never will be). Try accessing `this.foo` initialized in the constructor.

**Comment by shekohex** [comment]
@gremo  i think it's now should be fixed
Thanks again ❤

**Comment by bashleigh** [comment]
This is now published `v1.2.3`
